### PR TITLE
Vue - Vuex

### PR DESCRIPTION
**Pull request by renzowolox**
## Summary
* Store added
* Refactor books view

## Screenshots
image

## Trello Card
[Vuex](https://trello.com/c/lNptlYLN/19-vuex)

### CODE REVIEW

File: vue-training/src/store/index.js [hunk 1]
```diff
@@ -1,14 +1,26 @@
 import Vue from 'vue'
 import Vuex from 'vuex'
+import { getBooks } from '@/services/BooksService'
 
 Vue.use(Vuex)
 
 export default new Vuex.Store({
```
Comment (rsuescun) [inline vue-training/src/store/index.js:7]: Add module for books state

```
/store
|--index.js
|--/modules 
```

Check this https://vuex.vuejs.org/guide/modules.html#module-local-state

File: vue-training/src/views/Books.vue [hunk 2]
```diff
@@ -21,17 +21,11 @@ export default {
...
+    this.$store.dispatch('getBooksList')
   },
-  data () {
-    return {
-      bookList: []
-    }
-  }
+  computed: mapState({
+    books: 'bookList'
+  })
```
Comment (rsuescun) [inline vue-training/src/views/Books.vue]: When adding the books module, change this

```
  computed: {
    ...mapState({
      books: state => state.module.bookList
    })
  }
```

**Review by rsuescun — CHANGES_REQUESTED** [review]

File: vue-training/src/views/Books.vue (hunk 2 above)
Comment (rsuescun) [reply vue-training/src/views/Books.vue]: I don't see this change

Comment (renzowolox) [reply vue-training/src/views/Books.vue]: see it in the commitment below
[Vuex - module added](https://github.com/wolox-training/rt-react/pull/105/commits/5614a3dcf48a6319b594bbd982c532719cd87bb9)

File: vue-training/src/views/Books.vue [hunk 3]
```diff
@@ -21,17 +21,11 @@ export default {
...
+    this.$store.dispatch('books/getBooksList')
   },
-  data () {
-    return {
-      bookList: []
-    }
-  }
+  computed: mapState({
+    books: state => state.books.bookList
+  })
```
Comment (rsuescun) [inline vue-training/src/views/Books.vue]:
```suggestion
computed: {
  ...mapState({
    books: state => state.books.bookList
  })
}
```

**Review by rsuescun — CHANGES_REQUESTED** [review]

File: vue-training/src/views/Books.vue [hunk 4]
```diff
@@ -21,16 +21,12 @@ export default {
     NavBar
   },
   mounted () {
-    getBooks().then(response => {
-      if (response.ok) {
-        this.bookList = response.data
-      }
-    })
+    this.$store.dispatch('books/getBooksList')
```
Comment (rsuescun) [inline vue-training/src/views/Books.vue]: Use mapActions

```
  methods: {
    ...mapActions({
      nameAction: 'module/nameAction'
    })
  }
```

See https://vuex.vuejs.org/guide/actions.html#dispatching-actions-in-components

**Review by rsuescun — CHANGES_REQUESTED** [review]

**Review by rsuescun — APPROVED** [review]

**Pull request merged**